common/lfdemod.c

```diff
@@ -426,543 +308,410 @@ uint32_t manchesterEncode2Bytes(uint16_t datain) {
 
 //by marshmellow
 //encode binary data into binary manchester 
-int ManchesterEncode(uint8_t *BitStream, size_t size)
-{
-	size_t modIdx=20000, i=0;
+//NOTE: BitStream must have double the size available in memory to do the swap
+int ManchesterEncode(uint8_t *BitStream, size_t size) {
+	size_t modIdx=size, i=0;
 	if (size>modIdx) return -1;
```

Comment:
i patched this issue and made it better than it was.  ultimately later i think i may add a more array size safe function or a feed and destination array with proper size parameters to check to make sure we can't overrun the array.